### DISCUSSION TITLE
How to clone multiple repositories in workflow

### DISCUSSION THREAD

**Discussion by Mayankm13** [community/community#37820] [Actions] [answered]
My goal is to clone multiple repositories so that I can configure the Continuous Integration workflow in GitHub.

I am getting below error

Below is my workflow for PUSH
on: [push]
env:
WF_ENV: workflow environment variables
PEM_KEY: ${{ secrets.}}
jobs:
run-shell-command:
runs-on: [ 'self-hosted', 'linux', 'x64', 'docker' ]
steps:
- name: checking host details
run: |
pwd
uname -a
pwd
ls -lath
git --version
git config --list
echo "GITHUB_ACTOR: ${GITHUB_ACTOR}"
echo "GIT_REPOSITORY: ${GIT_REPOSITORY}"
echo ${PEM_KEY} > test.pem
cat test.pem
echo ${{ github.token }}

  - name: checking git config details 
    run: |
      pwd 
      git config  --list
      ls -lath   
      git clone https://${{ github.token }}@Ip/REPO/online-external-libs       
      pwd 
      ls -lath   
  - name: clone shared libs 
    uses: actions/checkout@v3
    with:
      token: ${{ github.token }}
      
  - name: checking host details 
    run: |
      pwd
      ls -alrth  
      cd Linux
      ls -ltrha

**Comment by airtower-luna** [comment, marked as answer]
The immediate issue seems to be that you put `ip` (I guess that's a placeholder for an actual IP address?) into the URL instead of the correct hostname. The IP is not listed in the certificate used for HTTPS, so the connection fails. Using the hostname should fix that problem.

However, your workflow would be a lot simpler with multiple calls to `actions/checkout`. There are [a variety of ways](https://github.com/actions/checkout#checkout-multiple-repos-side-by-side) to retrieve multiple repositories into your workspace. :slightly_smiling_face:

**Reply by Mayankm13** [reply]
In the issue description, I just replaced the actual IP address with the IP keyword. Also, I tried the suggested way, but I still got the following error


![image](https://user-images.githubusercontent.com/91568786/198901214-20640ae8-5829-4060-84d5-5881989f3f81.png)


name: first github workflow 

on: [push]
env: 
  WF_ENV: workflow envirnment variables
  PEM_KEY: ${{ secrets.APP_PEM_BADAM }}
jobs: 
  run-shell-command: 
    runs-on: [ 'self-hosted', 'linux', 'x64', 'docker' ]
    steps:
      - name: checking host details 
        run: |
          pwd
          uname -a
          pwd
          ls -lath 
          git --version
          git config  --list
          echo "GITHUB_ACTOR:  ${GITHUB_ACTOR}"
          echo "GIT_REPOSITORY: ${GIT_REPOSITORY}"
          echo ${PEM_KEY} >  test.pem
          cat test.pem


      - name: checkout
        uses: actions/checkout@v2
        with:
          submodules: true
          token: ${{ secrets.PAT_BADAM }}
      - name: checking git config details 
        run: |
          pwd 
          git config  --list
          ls -lath         
      - name: checking host details 
        run: |
          pwd
          ls -alrth  
          cd utils-common
          pwd
          ls -altrh 
      - name: clone shared libs 
        uses: actions/checkout@v2
        with:
          repository: github.oneeuronet.com:Euronet-REN/renaissance-online-external-libs.git
          ref: master
          token: ${{ secrets.PAT_BADAM }}
          submodules: true



      - name: checking host details 
        run: |
          pwd
          ls -alrth  
          cd Linux
          ls -ltrha

**Reply by airtower-luna** [reply]
> In the issue description, I just replaced the actual IP address with the IP keyword.

Then try using the correct hostname instead. I notice the hostname mentioned in the error message in your first post doesn't match the one in this post. Could be a config problem, so simply virtual hosts on one machine not working without the proper hostname.

For `actions/checkout`, *don't* include the hostname in the `repository` input. The [usage documentation](https://github.com/actions/checkout#usage) mentions a dedicated `github-server-url` parameter.

Also, `actions/checkout@v2` is outdated, you should be using `v3`.

So your step should be something like:

```yaml
  - name: clone shared libs 
    uses: actions/checkout@v3
    with:
      github-server-url: 'https://github.oneeuronet.com'
      repository: Euronet-REN/renaissance-online-external-libs.git
      ref: master
      token: ${{ secrets.PAT_BADAM }}
      submodules: true
```

**Reply by Mayankm13** [reply]
Thank you so much, it worked, but it Deleted  the contents  of the first repo and then clone the second repo 

please check the below snap for your reference.

![image](https://user-images.githubusercontent.com/91568786/198902559-11d9f7cf-a678-4e65-b826-ce74c57a0b2f.png)

**Reply by airtower-luna** [reply]
Did you actually update `actions/checkout` to `v3` ([`v3.1.0`](https://github.com/actions/checkout/releases/tag/v3.1.0) is the latest version)? That input is [definitely defined](https://github.com/actions/checkout/blob/1f9a0c22da41e6ebfa534300ef656657ea2c6707/action.yml#L74-L76), so my guess is that you're still on an older one.

**Reply by Mayankm13** [reply]
@airtower-luna, Thanks for your help so far. Now that I'm using the new update actions/checkout, I'm getting the following warning.

![image](https://user-images.githubusercontent.com/91568786/198903498-ff7f0a55-8358-4c4a-9437-f58fc4e64eb9.png)

![image](https://user-images.githubusercontent.com/91568786/198903556-035ef3c5-c461-4bf0-946c-d9866043bec5.png)

**Reply by airtower-luna** [reply]
If you're sure the action is up to date (`v3.1.0` is pretty recent, and I noticed the input was only added there) you might want to raise an issue in the `actions/checkout` repository. But I'd really double check, if you use any older `v3` (e.g. a local copy on your GHES instance) it won't be there. :sweat_smile:

**Reply by Mayankm13** [reply]
please find below my workflow, I am using actions/checkout@v3

 The objective is to clone EuronetsoftwareRenaissanceOnline and then clone the second repo on the same path to run my build script.


`name: first github workflow 

on: [push]
env: 
  WF_ENV: workflow envirnment variables
  PEM_KEY: ${{ secrets.APP_PEM_BADAM }}
jobs: 
  run-shell-command: 
    runs-on: [ 'self-hosted', 'linux', 'x64', 'docker' ]
    steps:
      - name: checking host details 
        run: |
          pwd
          uname -a
          pwd
          ls -lath 
          git --version
          git config  --list
          echo "GITHUB_ACTOR:  ${GITHUB_ACTOR}"
          echo "GIT_REPOSITORY: ${GIT_REPOSITORY}"
          echo ${PEM_KEY} >  test.pem
          cat test.pem


      - name: checkout
        uses: actions/checkout@v3
        with:
          submodules: true
          token: ${{ secrets.PAT_BADAM }}
      - name: checking git config details 
        run: |
          pwd 
          git config  --list
          ls -lath         
      - name: checking host details 
        run: |
          pwd
          ls -alrth  
          cd utils-common
          pwd
          ls -ltrha
      - name: clone shared libs 
        uses: actions/checkout@v3
        with:
          github-server-url: 'https://github.oneeuronet.com'
          repository: Euronet-REN/renaissance-online-external-libs.git
          ref: master
          token: ${{ secrets.PAT_BADAM }}
          submodules: true
      
      - name: checking host details 
        run: |
          pwd
          ls -alrth  
          cd linux
          pwd
          ls -ltrh

          `

**Reply by airtower-luna** [reply]
Are you absolutely sure that you get `v3.1.0` if you call `v3`? The action should be retrieved from your local server, and I don't know if that's up to date with `github.com`. Or directly try `actions/checkout@v3.1.0`, that will fail directly if you don't have that version. :wink:

**Reply by Mayankm13** [reply]
@airtower-luna  When I  try direct actions/checkout@v3.1.0, Can you please let me know how to fix direct actions/checkout@v3.1.0 failing?

![image](https://user-images.githubusercontent.com/91568786/199014412-f349b7b9-291c-4e37-ba92-f73f2510c72f.png)

**Reply by airtower-luna** [reply]
Talk to whoever is in charge of the `actions/checkout` repository on your GHES instance. They'll have to pull in the updates from GitHub.

**Reply by Mayankm13** [reply]
Thankyou it's resloved

**Reply by airtower-luna** [reply]
Excellent, glad I could help! Please don't forget to mark the answer. :slightly_smiling_face:
